### PR TITLE
Added Prowlarr, updated Dozzle + Uptime-Kuma with socket-proxy option

### PR DESCRIPTION
**Pull request by di5cord20**
Hope this helps others.  

I didn't know how the syntax for OR in the links secion.  Please update if needed.
Currently only checks if sonarr included to embed the ",".  This could cause a problem if someone doesn't include sonarr when creating the container (not a problem for me as I did include sonarr).

`        "links": [
          {if Context.sonarr}
            "{Context.sonarr-name}"
          {/if}
          {if Context.radarr}
            {if Context.sonarr}
              ,
            {/if}
            "{Context.radarr-name}"
          {/if}
          {if Context.lidarr}
            {if Context.sonarr}
              ,
            {/if}
            "{Context.lidarr-name}"
          {/if}
          {if Context.readarr}
            {if Context.sonarr}
              ,
            {/if}
            "{Context.readarr-name}"
          {/if}
          {if Context.downloader}
            {if Context.sonarr}
            ,
            {/if}
            "{Context.downloader-name}"
          {/if}    
        ]`

### CODE REVIEW

**Comment by di5cord20** [comment]
Added a 2nd commit enhancing both the Dozzle and Uptime-Kuma apps by adding an option to use a socket-proxy vs. binding the docker.sock directly.  I tested locally but encourage others to do so as well. 

1. Dozzle and Uptime-Kuma will now ask if you want to use a socket-proxy (which is better than bind the docker.sock directly).
2. You will then ask if you already have one. 
- If yes, you need to select the container
- If no, the socket-proxy container/app will be created

3. A link will be created between Dozzle/Uptime-Kuma to the socket-proxy container/app.

**Comment by di5cord20** [comment]
Hi @azukaar , not sure if you had a chance to review.  Happy to make updates if needed.

File: servapps/Prowlarr/cosmos-compose.json [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+          },
+          "AuthEnabled": true
+          {if Context.adminOnly}
+           , "AuthAdmin": true
+          {/if}
+        }
+      ],
+      "links": [
+        {if Context.sonarr}
+          "{Context.sonarr-name}"
```
Comment (azukaar) [inline servapps/Prowlarr/cosmos-compose.json:95]: here, if you dont use Sonarr then you end up starting the array with a coma

**Comment by azukaar** [comment]
I reviewed it, there's a small issue with it, but I am not exactly sure how to fix it @di5cord20 that's why I need to do some additional testing. But basically the issue is that  when you select different combination of tools (like Radarr or Sonarr) you might end up in a template where the number of coma is wrong
I will think about this more and ping you when I have a  solution!

**Comment by di5cord20** [comment]
Yup, that's the issue I mentioned in my commit.  I couldn't figure out syntax for an OR operator which could be used for the initial check before adding a the comma.  In non-technical terms, check if any of the potential containers exist, if yes, add a comma and move to the links section to add the required links.

This impacts the Prowlarr app.  For Bazarr and Uptime-Kuma, I used a slightly different approach to circumvent by asking a few yex/no (checkbox) questions to determine which code ultimately is included in the compose.

**Comment by azukaar** [comment]
This is the template library I am using : https://github.com/gsf/whiskers.js/

**Comment by di5cord20** [comment]
I'll take a look and try to think of a solution.

Question: what does the "minVersion" line before "services" represent?  I see it across all the market apps with different values but have no idea what it's purpose is.  I assume the value matters so just want to use accordingly.  Thanks

**Comment by azukaar** [comment]
Basically Cosmos wont let you install it if your Cosmos version is below the minVersion, to avoid trying to install it on outdated instances that might not support the features you need. I would say rule of thumb is just to set it to be your current version just to be sure

**Pull request closed without merging**